index.js

```diff
@@ -28,7 +28,7 @@ function ghGot(path, opts) {
 	}
 
 	var endpoint = env.GITHUB_ENDPOINT ? env.GITHUB_ENDPOINT.replace(/[^/]$/, '$&/') : opts.endpoint;
-	var url = (endpoint || 'https://api.github.com/') + path;
+	var url = /https?/.test(path) ? path : endpoint + path;
```

Comment:
Is this enough? At first I was thinking to use [url.parse](https://nodejs.org/api/url.html#url_url_parse_urlstr_parsequerystring_slashesdenotehost) and check if the `host` property is empy. But this seemed faster as it doesn't have to parse the url every time.